File: Bai_7/main.py
def check_SHT(n):
    S = 0
    for i in range(1, n):
        if n % i == 0:
            S += i 

    if S == n:
        return True  
    return False  

def main():
    n = int(input("nhap so nguyen duong ma muon kiem tra: "))
    result = check_SHT(n)
    if result == True:
        print(n, "la mot so hoan thien!")
    else:
        print(n, "la mot so khong hoan thien!")

if __name__ == "__main__":
    main()

File: Bai_7/main.cpp
#include <iostream>
using namespace std;

int TongUocSo(int &n)
{
    int sum(0);
    for(int i = 1; i < n; i++)
    {
        if(n % i == 0)
            sum += i;
    }
    return sum;
}

int main()
{
    int n;
    cout << "Nhap so nguyen duong ma muon kiem tra: ";
    cin >> n;

    if(TongUocSo(n) == n)
        cout << n << " la mot so hoan thien!" << endl;
    else    
        cout << n << " khong phai la so hoan thien!" << endl;

    
    return 0;
}